qa/pull-tester/rpc-tests.py

```diff
@@ -184,26 +176,39 @@
     'replace-by-fee.py',
 ]
 
+ALL_SCRIPTS = BASE_SCRIPTS + ZMQ_SCRIPTS + EXTENDED_SCRIPTS
 
 def runtests():
-    test_list = []
-    if '-extended' in opts:
-        test_list = testScripts + testScriptsExt
-    elif len(opts) == 0 or (len(opts) == 1 and "-win" in opts):
-        test_list = testScripts
+    # Build list of tests
+    if len(tests) != 0:
+        # Individual tests have been specified. Run specified tests that exist
+        # in the ALL_SCRIPTS list. Accept the name with or without .py extension.
+        test_list = [t for t in ALL_SCRIPTS if
+                (t in tests or re.sub(".py$", "", t) in tests)]
```

Comment:
This is required if the user specifies a test with the .py suffix, eg:

`rpc-tests.py rpcnamedargs.py`